kythe/java/com/google/devtools/kythe/analyzers/java/KytheTreeScanner.java

```diff
@@ -290,12 +290,21 @@ public JavaNode visitIdent(JCIdent ident, TreeContext owner) {
     if (ident.sym == null) {
       return emitDiagnostic(ctx, "missing identifier symbol", null, null);
     }
-    EdgeKind edgeKind = EdgeKind.REF;
+
+    JavaNode node = null;
     if (ident.sym instanceof ClassSymbol && ident == owner.getNewClassIdentifier()) {
       // Use ref/id edges for the primary identifier to disambiguate from the constructor.
-      edgeKind = EdgeKind.REF_ID;
+      node = emitSymUsage(ctx, ident.sym, EdgeKind.REF_ID);
+    } else {
+      RefType refType = getRefType(owner.getTree(), ident.sym, ident.pos);
+      if (refType == RefType.READ || refType == RefType.READ_WRITE) {
```

Comment:
So if it's a READ_WRITE, we emit both the ref and the ref/write? Is there a test for this?

Reply:
Yes, the follow tests this:
```
//- @a ref A
//- @a ref/writes A
a += 1;
```